nifi-extension-bundles/nifi-standard-bundle/nifi-standard-rules/src/main/java/org/apache/nifi/flowanalysis/rules/RestrictLoadBalancing.java

```diff
@@ -0,0 +1,266 @@
+            .required(false)
+            .allowableValues("true", "false")
+            .defaultValue("true")
+            .addValidator(StandardValidators.BOOLEAN_VALIDATOR)
+            .build();
+    public static final PropertyDescriptor ALLOW_CONTENT_COMPRESSION = new PropertyDescriptor.Builder()
+            .name("Allow Content Compression")
+            .description("If set to true, connections which are configured to load balance may compress attributes and content. "
+                    + "This property is ignored if the only allowed Load Balance Strategy is 'Do not load balance'. "
+                    + "When set to true, 'Allow Attribute Compression' must also be set to true")
+            .required(false)
+            .allowableValues("true", "false")
+            .defaultValue("true")
+            .addValidator(StandardValidators.BOOLEAN_VALIDATOR)
+            .build();
+
+    private static final List<PropertyDescriptor> PROPERTY_DESCRIPTORS = List.of(
+            ALLOW_DO_NOT_LOAD_BALANCE,
+            ALLOW_PARTITION,
+            ALLOW_ROUND_ROBIN,
+            ALLOW_SINGLE_NODE,
+            ALLOW_ATTRIBUTE_COMPRESSION,
+            ALLOW_CONTENT_COMPRESSION
+    );
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return PROPERTY_DESCRIPTORS;
+    }
+
+    @Override
+    protected Collection<ValidationResult> customValidate(ValidationContext validationContext) {
+        final List<ValidationResult> results = new ArrayList<>();
+
+        // For this flow analysis rule to be valid, at least one load balancing strategy must be allowed
+        if (!validationContext.getProperty(ALLOW_DO_NOT_LOAD_BALANCE).asBoolean()
+                && !isLoadBalancingAllowed(validationContext)) {
+            results.add(new ValidationResult.Builder()
+                    .subject("Load Balance Strategy Properties")
+                    .valid(false)
+                    .explanation(CONFIGURE_STRATEGY_ERROR_MESSAGE)
+                    .build());
+        }
+
+        // The only forms of compression are "attributes only" and "both attributes and content".
+        // Therefore, if content compression is allowed, attribute compression must also be allowed.
+        // It is also valid to have only attribute compression or no compression at all allowed.
+        // In addition, this is only relevant if a load balance strategy is allowed, i.e. if the only allowed strategy is "Do not load balance",
+        // then any combination of compression is allowed because it will be ignored.
+        if (isLoadBalancingAllowed(validationContext)
+                && validationContext.getProperty(ALLOW_CONTENT_COMPRESSION).isSet() && validationContext.getProperty(ALLOW_CONTENT_COMPRESSION).asBoolean()
+                && validationContext.getProperty(ALLOW_ATTRIBUTE_COMPRESSION).isSet() && !validationContext.getProperty(ALLOW_ATTRIBUTE_COMPRESSION).asBoolean()) {
```

Comment:
Since both ALLOW_CONTENT_COMPRESSION and ALLOW_ATTRIBUTE_COMPRESSION have allowable values, they will always be set so I don't think you need the `isSet()` check.